File: sdk/c/src/ssl.c
#include "hitechcloud/ssl.h"
#include <stdlib.h>


/* GET /api/certificate - List all SSL certificates */
char *hc_ssl_list(hc_http_client_t *client) {
    return hc_get(client, "/api/certificate", NULL);
}

/* GET /api/certificate/{id} - Return certificate details */
char *hc_ssl_get(hc_http_client_t *client) {
    return hc_get(client, "/api/certificate/{id}", NULL);
}

/* GET /api/certificate/{id}/crt - Download X.509 certificate */
char *hc_ssl_download(hc_http_client_t *client) {
    return hc_get(client, "/api/certificate/{id}/crt", NULL);
}

/* GET /api/certificate/order - List available certificates for purchase */
char *hc_ssl_list_available(hc_http_client_t *client) {
    return hc_get(client, "/api/certificate/order", NULL);
}

/* POST /api/certificate/order - Order new certificate */
char *hc_ssl_order(hc_http_client_t *client) {
    return hc_post(client, "/api/certificate/order", NULL);
}

/* GET /api/certificate/order/{product_id}/software - List server software */
char *hc_ssl_list_server_software(hc_http_client_t *client) {
    return hc_get(client, "/api/certificate/order/{product_id}/software", NULL);
}

File: sdk/r/R/ssl.R
#' SSL Certificates: list, order, download certificates
#' @param http HttpClient object
#' @return Resource object
#' @export
SSLResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "SSLResource"
  env
}

#' GET /api/certificate - List all SSL certificates
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
list <- function(self, ...) {
  self$http$get("/api/certificate", list(...))
}

#' GET /api/certificate/{id} - Return certificate details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
get <- function(self, ...) {
  self$http$get("/api/certificate/{id}", list(...))
}

#' GET /api/certificate/{id}/crt - Download X.509 certificate
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
download <- function(self, ...) {
  self$http$get("/api/certificate/{id}/crt", list(...))
}

#' GET /api/certificate/order - List available certificates for purchase
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listAvailable <- function(self, ...) {
  self$http$get("/api/certificate/order", list(...))
}

#' POST /api/certificate/order - Order new certificate
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
order <- function(self, ...) {
  self$http$post("/api/certificate/order", list(...))
}

#' GET /api/certificate/order/{product_id}/software - List server software
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listServerSoftware <- function(self, ...) {
  self$http$get("/api/certificate/order/{product_id}/software", list(...))
}

